File: nodejs/src/driver.js
var Request = require('./request.js');
var Action = require('./action.js');
var Segment = require('./segment.js');
var Target = require('./target.js');
var Send = require('./send.js');
var Field = require('./field.js');
var Import = require('./import.js');
var configFile = require('./config.json');

function Driver(config) {
  if (!config) {
    throw new TypeError('Missing parameter');
  }

  this._xkey = config.xkey;
  this._url = config.baseUrl || configFile.baseUrl;
  this._request = new Request(this._xkey);
}

function defineEntity(name, Entity) {
  Object.defineProperty(Driver.prototype, name, {
    get: function() {
      var entity = new Entity(this);
      Object.defineProperty(this, name, {
        value: entity
      });
      return entity;
    }
  });
}

defineEntity('action', Action);
defineEntity('segment', Segment);
defineEntity('target', Target);
defineEntity('send', Send);
defineEntity('field', Field);
defineEntity('import', Import);

module.exports = Driver;

File: java/src/main/java/route/Driver.java
package route;

import org.json.JSONObject;

/**
 *  Driver class that contains all availables routes
 */
public class Driver {

    protected String _xkey;
    protected String _url;
    protected Connection _connection;

    public Segment segment;
    public Action action;
    public Target target;
    public Import imports;
    public Send send;
    public Field field;

    /**
     * Constructor
     *
     * @param config    contains configuration for driver
     * @throws IllegalArgumentException
     */
    public Driver(JSONObject config) throws IllegalArgumentException {
        if (config == null) {
            throw new IllegalArgumentException("Parameter cannot be null");
        }

        this._url = config.has("url") ? config.getString("url") : "http://v8.mailperformance.com/";
        this._xkey = config.getString("xkey");
        this._connection = new Connection(this._xkey, this._url);
        this.segment = new Segment(this._connection);
        this.action = new Action(this._connection);
        this.target = new Target(this._connection);
        this.imports = new Import(this._connection);
        this.send = new Send(this._connection);
        this.field = new Field(this._connection);
    }

    public String getUrl() {
        return this._url;
    }

    public String getXKey() {
        return this._xkey;
    }
}